index.rst

```diff
@@ -1,29 +1,299 @@
+
+The registry service knows all of the information needed for that MySQL table, and we don't want to have to maintain that information in multiple places.
+Therefore, the registry service, after it assembles the XML document that it will serve from the capabilities endpoints of the services available in that environment, should replace the registry table in the MySQL service with a SQL version of the same data.
+From there, it can be queried by the TAP service similar to the TAP schema table.
+
+.. _tap-schema:
+
+TAP schema
+==========
+
+The TAP service itself can be queried for its own schema by querying the ``TAP_SCHEMA`` schema.
+This is specified in the TAP_ standard.
+We therefore only need to populate the ``TAP_SCHEMA`` table and serve it through the TAP service.
+
+.. _TAP: https://www.ivoa.net/documents/TAP/
+
+Observation data will be served by a separate ObsTAP_ service that needs its own ``TAP_SCHEMA`` table containing only the schema supported by that TAP service.
+See :dmtn:`236` for more information about the ObsTAP design.
+
+.. _ObsTAP: https://www.ivoa.net/documents/ObsCore/
+
+As discussed in :ref:`VO services <vo-implementation>`, we will also want to serve RegTAP_ data.
+The schema for that data will also need to be part of ``TAP_SCHEMA``.
+
+Implementation
+--------------
+
+Schemas for the Rubin Science Platform are managed using Felis_ from metadata stored in the `sdm_schemas repository`_.
+As part of the build process of that repository, Felis generates MySQL tables holding the schema information and bundles them into a Docker image for a MySQL server that contains only the ``TAP_SCHEMA`` table.
+This server is deployed in Phalanx_ as the tap-schema application, and the TAP server is configured to point to it.
+
+.. _Felis: https://felis.lsst.io/
+.. _sdm_schemas repository: https://github.com/lsst/sdm_schemas
+
+Currently, that schema does not contain RegTAP_ schema, since we're not serving RegTAP data.
+When we start doing so, we can incorporate that schema into the ``TAP_SCHEMA`` table generated from the `sdm_schemas repository`_.
+In order to serve the RegTAP data itself out of the same MySQL service used for ``TAP_SCHEMA``, we may have to make additional modifications to the TAP service to add additional backend data sources.
+
+How to provide ``TAP_SCHEMA`` for the ObsTAP service is not yet clear.
+The simplest approach will be if the PostgreSQL backend for ObsTAP can itself provide a ``TAP_SCHEMA`` table as well.
+Then, the ObsTAP service can use a single backend for all data.
+
+If that's not possible, we will need to create a separate obstap-schema Phalanx service and build a separate set of MySQL images that contain the ObsCore schemas instead of the schemas for the regular TAP service.
```

Comment:
I can't think of why this would not be possible. If ObsTAP back end is postrgres, it should surely just have the schema. Now if ObsTAP back end is Qserv, maybe there's a question there, but we should just resolve it with GPDF now and write in the final answer.

Reply:
I did talk about this with GPDF before writing this tech note, and he seemed to think that `TAP_SCHEMA` for ObsTAP would continue to be in a separate database generated from sdm_schemas, but I admit I didn't ask him directly whether it would be possible to roll it into the ObsTAP database. I'll bring this up in the Science Platform meeting today if we're having it.